kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.java

```diff
@@ -85,116 +83,64 @@
     private final Boolean cascading;
 
   @Override
-  public List<HasMetadata> waitUntilReady(final long amount, final TimeUnit timeUnit) throws InterruptedException {
-    List<HasMetadata> items = acceptVisitors(asHasMetadata(item, true), visitors);
-    if (items.isEmpty()) {
-      return Collections.emptyList();
-    }
-
-    final List<HasMetadata> result = new ArrayList<>();
-    final List<HasMetadata> itemsWithConditionNotMatched = new ArrayList<>(items);
-    final int size = items.size();
-    final ExecutorService executor = Executors.newFixedThreadPool(size);
-
-    try {
-      final CountDownLatch latch = new CountDownLatch(size);
-      for (final HasMetadata meta : items) {
-        final ResourceHandler<HasMetadata, ?> h = handlerOf(meta);
-        if (!executor.isShutdown()) {
-          executor.submit(() -> {
-            try {
-              result.add(h.waitUntilReady(client, config, meta.getMetadata().getNamespace(), meta, amount, timeUnit));
-            } catch (InterruptedException | IllegalArgumentException interruptedException) {
-              // We may get here if waiting is interrupted or resource doesn't support concept of readiness.
-              // We don't want to wait for items that will never become ready
-              // Skip that resource then.
-              LOGGER.info("{} {} does not support readiness. skipping..", meta.getKind(), meta.getMetadata().getName());
-              latch.countDown();
-            } catch (IllegalStateException t) {
-              logAsNotReady(t, meta);
-            } finally {
-              // Resource got ready and was returned properly
-              latch.countDown();
-            }
-          });
-        }
-      }
-      latch.await(amount, timeUnit);
-      if (latch.getCount() == 0) {
-        return result;
-      } else {
-        throw new KubernetesClientTimeoutException(itemsWithConditionNotMatched, amount, timeUnit);
-      }
-    } finally {
-      executor.shutdown();
-    }
+  public List<HasMetadata> waitUntilReady(final long amount, final TimeUnit timeUnit) {
+    return waitUntilCondition(resource -> Objects.nonNull(resource) && getReadiness().isReady(resource), amount, timeUnit);
```

Comment:
The statement
```java
getReadiness().isReady(resource)
```
is not equivalent to
```java
handlerOf(resource).waitUntilReady(...)
```

However the former doesn't make much sense since the effective call of waitUntilRady should be that of the applicable *OperationsImpl (e.g. ServiceHandler->ServiceOperationsImpl), especially considering the ServiceTest. Are we sure the Endpoints are being queried after these changes?

Reply:
https://github.com/fabric8io/mockwebserver/issues/62